src/deepsparse/pipeline.py

```diff
@@ -129,13 +130,20 @@ def __init__(
         input_shapes: List[List[int]] = None,
         alias: Optional[str] = None,
         context: Optional[Context] = None,
+        threadpool: Optional[ThreadPool] = None,
     ):
         self._model_path_orig = model_path
         self._model_path = model_path
         self._engine_type = engine_type
         self._alias = alias
-
         self.context = context
+        self._threadpool = threadpool
+
+        if self._threadpool and not isinstance(self._threadpool, ThreadPool):
```

Comment:
This seems overly strict to me. I think I'd either cope with loose duck typing or add an abstract class to use as an interface and require that whatever match the interface instead of being a specific class implementation.

Reply:
Now we check if a `ThreadPoolExecutor()` is passed